lib/vtls/nss.c

```diff
@@ -1600,6 +1600,10 @@ static void nss_close(struct Curl_easy *data, struct connectdata *conn,
     conn->sock[sockindex] = CURL_SOCKET_BAD;
   }
 
+  /* disable NPN + APLN to make the HandshakeCallback return early if it
+     gets called from the PR_Recv() within close_one() */
+  conn->bits.tls_enable_npn = conn->bits.tls_enable_alpn = FALSE;
+
```

Comment:
imo this is too hacky and would be better solved in the handshake callback, like

~~~diff
diff --git a/lib/vtls/nss.c b/lib/vtls/nss.c
index 9e30143..6317f00 100644
--- a/lib/vtls/nss.c
+++ b/lib/vtls/nss.c
@@ -882,8 +882,10 @@ static void HandshakeCallback(PRFileDesc *sock, void *arg)
        !memcmp(ALPN_HTTP_1_1, buf, ALPN_HTTP_1_1_LENGTH)) {
       conn->negnpn = CURL_HTTP_VERSION_1_1;
     }
-    Curl_multiuse_state(data, conn->negnpn == CURL_HTTP_VERSION_2 ?
-                        BUNDLE_MULTIPLEX : BUNDLE_NO_MULTIUSE);
+    if(conn->bundle) {
+      Curl_multiuse_state(data, conn->negnpn == CURL_HTTP_VERSION_2 ?
+                          BUNDLE_MULTIPLEX : BUNDLE_NO_MULTIUSE);
+    }
   }
 }
 
~~~

Reply:
sure, I'm okay with that, with a big comment block explaining the reasoning. I'll amend